src/utils/mtev_lfu.c

```diff
@@ -220,12 +236,23 @@ touch_lfu_cache_no_lock(mtev_lfu_t *c, struct lfu_entry *e)
     struct lfu_cache_entry *bucket = e->frequency_list_head;
     struct lfu_cache_entry *next_bucket = STAILQ_NEXT(bucket, list_entry);
 
+    mtev_boolean need_new_bucket = mtev_false;
+
     if (next_bucket == NULL ||
         next_bucket == STAILQ_FIRST(&c->lfu_frequency_list) ||
         next_bucket->frequency != bucket->frequency + 1) {
-      next_bucket = new_frequency_bucket_no_lock(c, bucket->frequency + 1, bucket);
+      need_new_bucket = mtev_true;
+    }
+    struct lfu_cache_entry *empty = remove_from_frequency_list_no_lock(c, e);
+    if (need_new_bucket) {
+      if (empty != NULL) {
+        next_bucket = empty;
+        next_bucket->frequency = bucket->frequency + 1;
```

Comment:
Just reading this, but isn't `next_bucket` == `bucket`? and wasn't `bucket->frequency` zero'd out `remove_from_frequency_list_no_lock`? need to cache the frequency, I think, not re-use it from the bucket.